src/main/kotlin/io/omnitrade/client/OmnitradeClient.kt

```diff
@@ -0,0 +1,407 @@
+package io.omnitrade.client
+
+import io.omnitrade.auth.Auth
+import io.omnitrade.di.Dependencies
+import io.omnitrade.enumerator.OrderBy
+import io.omnitrade.enumerator.deposit.DepositState
+import io.omnitrade.enumerator.http.HttpMethod
+import io.omnitrade.enumerator.k.Period
+import io.omnitrade.enumerator.market.MarketType
+import io.omnitrade.enumerator.order.OrderDepth
+import io.omnitrade.enumerator.order.OrderSide
+import io.omnitrade.enumerator.order.OrderState
+import io.omnitrade.enumerator.order.OrderType
+import io.omnitrade.exception.OmnitradeNoAuthKeysException
+import io.omnitrade.exception.OmnitradeRequestException
+import io.omnitrade.ext.toQueryStringArray
+import io.omnitrade.model.*
+import io.omnitrade.repository.OmnitradeRepository
+import org.kodein.di.Kodein
+import org.kodein.di.generic.M
+import org.kodein.di.generic.instance
+import retrofit2.HttpException
+import retrofit2.Response
+import java.util.concurrent.CompletableFuture
+
+class OmnitradeClient private constructor(
+  private var accessKey: String = "",
+  private var secretKey: String = "",
+  endpoint: String = "https://omnitrade.io/",
+  timeout: Long = 60,
+  dependencies: Kodein = Kodein { import(Dependencies.clientModule) }
+) {
+  companion object {
+    private const val API_VERSION = "/api/v2"
+  }
+
+  object Builder {
+    private var accessKey: String = ""
+    private var secretKey: String = ""
+    private var endpoint: String = "https://omnitrade.io/"
```

Comment:
Looks like we're repeating this from line 29, is it possible to reuse it?

Reply:
Sure! It makes sense. Made a const to be used for them both.